### PR TITLE
YDA-5747: Script for granting read members access to vault packages

### PR DESCRIPTION
**Pull request by claravox**
Script to grant read members of research groups read access to vault packages that have already been published.

### CODE REVIEW

File: vault.py [hunk 1]
```diff
@@ -1135,6 +1136,135 @@ def set_vault_permissions(ctx, coll, target):
...
+        if verbose:
+            modes.append("verbose")
+        log.write(ctx, "Running grant_readers_vault_access in {} mode.".format((" and ").join(modes)))
+
+    zone = user.zone(ctx)
+
+    # Get the group names
+    userIter = genquery.row_iterator(
+        "USER_GROUP_NAME",
+        "USER_TYPE = 'rodsgroup' AND USER_ZONE = '{}' AND USER_GROUP_NAME like 'research-%'".format(zone),
```
Comment (claravox) [inline vault.py:1251]: Should this include more groups?

**Comment by claravox** [comment]
The code isn't the most efficient, is that ok? Can either use more efficient algorithm or always set the vault permissions regardless of whether readers already have access.

File: vault.py (hunk 1 above)
Comment (lwesterhof) [reply vault.py:1251]: No, only needed for research groups.

**Comment by lwesterhof** [comment]
> The code isn't the most efficient, is that ok? Can either use more efficient algorithm or always set the vault permissions regardless of whether readers already have access.

Not a problem I think, we only need to run this once to fix the current inconsistencies.

**Review by lwesterhof — APPROVED** [review]
LGTM!

**Pull request merged**